src/order/ideal.lean

```diff
@@ -87,7 +97,7 @@ instance : partial_order (ideal P) := partial_order.lift coe ext
 
 /-- A proper ideal is one that is not the whole set.
     Note that the whole set might not be an ideal. -/
-class proper (I : ideal P) : Prop := (nuniv : (I : set P) ≠ set.univ)
+@[mk_iff] class proper (I : ideal P) : Prop := (ne_univ : (I : set P) ≠ set.univ)
```

Comment:
What does `mk_iff` do, out of curiosity?

Reply:
It can automatically generate `iff` lemmas for structures. See the [doc page](https://leanprover-community.github.io/mathlib_docs/attributes.html#mk_iff) and some of my resolved suggestions above.